src/App.tsx

```diff
@@ -1,24 +1,332 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoItem } from './components/TodoItem';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Error } from './components/Error';
 
-const USER_ID = 0;
+const USER_ID = 11616;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [errorWarning, setErrorWarning] = useState('');
+  const [pageIsLoaded, setPageIsLoaded] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdatingId, setIsUpdatingId] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState(0);
+  const [editQuery, setEditQuery] = useState('');
+  const [editQueryPrev, setEditQueryPrev] = useState('');
+
+  const input = document
+    .querySelector<HTMLInputElement>('.todoapp__new-todo');
+
+  const handleErrorSet = (errMessage: string) => {
+    setErrorWarning(errMessage);
+    setTimeout(() => {
+      setErrorWarning('');
+    }, 3000);
+  };
+
+  const closeError = () => {
+    setErrorWarning('');
+  };
+
+  const loadTodos = async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+      setPageIsLoaded(true);
+    } catch {
+      handleErrorSet('load-todo');
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    input?.focus();
+  }, [input, pageIsLoaded, todos]);
+
+  const visibleTodos = (filterBy: string) => {
+    if (todos) {
+      switch (filterBy) {
+        case 'all':
+          return todos;
+        case 'active':
+          return todos.filter(todo => (
+            !todo.completed
+          ));
+        case 'completed':
+          return todos.filter(todo => (
+            todo.completed
+          ));
+        default: return todos;
+      }
+    }
+
+    return todos;
+  };
+
+  const handleFormSubmit = async () => {
+    if (query.trim().length < 1) {
+      handleErrorSet('title-empty');
+
+      return;
+    }
+
+    const newTodo = {
+      id: todos.length > 0 ? todos[todos.length - 1].id + 1 : 1,
+      userId: USER_ID,
+      title: query.trim(),
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      setPageIsLoaded(false);
+      await addTodo('/todos', newTodo)
+        .then(() => {
```

Comment:
mixing async await and then().catch() syntax is not a good idea.
Choose one of them and be consistent.